### PR TITLE
Create wallet ID constants file

### PR DESCRIPTION
**Pull request by gregfromstl**
Doesn't cover all cases, will incrementally replace uses of wallet IDs with these constants

---

## PR-Codex overview
This PR updates wallet constants and uses them consistently across the codebase for popular wallets like Coinbase and MetaMask.

### Detailed summary
- Updated wallet constants for Coinbase, MetaMask, Rainbow, and Zerion
- Replaced hardcoded wallet IDs with constants in various files
- Improved consistency and readability in wallet-related code

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: packages/thirdweb/src/wallets/create-wallet.ts [hunk 1]
```diff
@@ -62,9 +63,9 @@ export function createWallet<const ID extends WalletId>(
      * COINBASE WALLET VIA SDK
      * -> if no injected coinbase found, we'll use the coinbase SDK
      */
-    case "com.coinbase.wallet": {
+    case COINBASE: {
       return coinbaseWalletSDK(
-        creationOptions as CreateWalletArgs<"com.coinbase.wallet">[1],
+        creationOptions as CreateWalletArgs<typeof COINBASE>[1],
```
Comment (joaquim-verges) [inline packages/thirdweb/src/wallets/create-wallet.ts:68]: maybe we just do `typeof X` everywhere rather than having another const

**Comment by MananTank** [comment]
I'm not really sure if it's useful to export a subset of wallet ids as constants. Could be more confusing than simply using the wallet id as string for library user

**Comment by gregfromstl** [comment]
> I'm not really sure if it's useful to export a subset of wallet ids as constants. Could be more confusing than simply using the wallet id as string for library user

The idea is mainly for internal use so we don't have the same manually typed string everywhere that could break with a typo. A user could theoretically import them but they shouldn't need to

**Comment by jnsdls** [comment]
> > I'm not really sure if it's useful to export a subset of wallet ids as constants. Could be more confusing than simply using the wallet id as string for library user
> 
> The idea is mainly for internal use so we don't have the same manually typed string everywhere that could break with a typo. A user could theoretically import them but they shouldn't need to

for internal use this makes sense to me, I don't think we should export them from the `/exports` at all (so they will not be publicly available, I don't think that is useful)

**Review by jnsdls — APPROVED** [review]

**Pull request merged**